### PR TITLE
User features

### PR DESCRIPTION
**Pull request by dsilberger**
Implements a user model using Sequelize, together with methods of interacting with the model, and a barebones userRoutes file. TBD: Supply functions referred to in userRoutes.

Also updates .gitignore to ignore config file.

### CODE REVIEW

File: server/db/models/usersModel.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    })
+    .catch(err => {
+      console.error(`Failed to create new user in database. Error: ${err}`);
+      return;
+    });
+};
+
+module.exports.getUser = function() {
+  User.findAll()
+    .then((err, data) => {
```
Comment (chengmuy) [inline server/db/models/usersModel.js]: Should the callback here just be (data) => {...}

File: server/db/models/usersModel.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+const { Sequelize, DataTypes } = require('sequelize');
+
+const sequelize = new Sequelize('budget', 'dsilberger', '', {
```
Comment (chengmuy) [inline server/db/models/usersModel.js]: I think "dsilberger" and "" should pull from config.js instead?

File: server/db/models/usersModel.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+module.exports.addUser = function(userObj) {
+  // userObj should be an object of the form
+  // { name: "some string", income: "some integer", familySize: "some integer" }
+  // id will be supplied by db.
+
+  let newUser = User.build(userObj);
+
+  newUser
+    .save()
+    .then((err, data) => {
```
Comment (chengmuy) [inline server/db/models/usersModel.js]: Should this callback be just (data) => {...} ?

File: server/db/models/usersModel.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  if (err) {
+    console.error(`An error occurred updating the Users table. Error: ${err}`);
+  } else {
+    console.log(`Users synced to model.`);
+  }
+});
+
+// User methods
+
+module.exports.addUser = function(userObj) {
```
Comment (chengmuy) [inline server/db/models/usersModel.js]: Consider destructuring syntax for self-documenting purposes

function({name, familysize, income})

User.build({name, familysize, income})

File: server/db/models/usersModel.js (hunk 1 above)
Comment (dsilberger) [reply server/db/models/usersModel.js]: Agree

File: server/db/models/usersModel.js (hunk 3 above)
Comment (dsilberger) [reply server/db/models/usersModel.js]: Agree

File: server/routing/usersRouter.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const usersRouter = require('express').Router();
+const userController = require('../db/controllers/usersController.js');
+
+usersRouter
+  .route('/sign-up')
```
Comment (chengmuy) [inline server/routing/usersRouter.js]: Let's call this endpoint /user or /users in the interests of being RESTful

**Comment by chengmuy** [comment]
It doesn't look like the endpoint was updated

File: server/routing/usersRouter.js (hunk 5 above)
Comment (chengmuy) [reply server/routing/usersRouter.js]: We'll merge this and address it in the next PR

**Pull request merged**